cli/cmd/delete.go

```diff
@@ -44,59 +45,48 @@ var deleteCmd = &cobra.Command{
 }
 
 func runDelete(cmd *cobra.Command, args []string) error {
-	ctx := context.WithValue(cmd.Context(), options.DryRunKey, dryRun)
 	stackName := args[0]
 
-	bootstrapConfig, err := config.NewBootstrapConfig(cmd)
+	happyClient, err := makeHappyClient(cmd, sliceName, stackName, []string{tag}, createTag)
```

Comment:
I traced the code to starting the worker pool, and see that `config.NewHappyConfig()` is called in a bunch of other places --  does it make sense to make the tfc worker pool global and create it globally so we won't miss another one of these locks? Or, perhaps, not synchronize if there's no pool?

Reply:
the only reason I did it there was because we need a context. Maybe I can find a place to grab the context for all commands

Reply:
see https://github.com/chanzuckerberg/happy/pull/1825/commits/08a6167162357198fcd8fdd5930438536529ff8d @alexlokshin-czi